# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/txmgr/nsITransactionManager.idl
# Commit: d3816f82c0b9
# Full Hash: d3816f82c0b9d4ceca0bb2ad177694164ed0921f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r 876b1031a6ae -r d3816f82c0b9 editor/txmgr/nsITransactionManager.idl
--- a/editor/txmgr/nsITransactionManager.idl	Fri Apr 03 01:38:22 2020 +0000
+++ b/editor/txmgr/nsITransactionManager.idl	Fri Apr 03 01:39:36 2020 +0000
@@ -31,18 +31,21 @@
    * stack is pruned or when the transaction manager is destroyed.
    * @param aTransaction the transaction to do.
    */
+  [can_run_script]
   void doTransaction(in nsITransaction aTransaction);
 
   /**
    * Pops the topmost transaction on the undo stack, calls its
    * undoTransaction() method, then pushes it on the redo stack.
    */
+  [can_run_script]
   void undoTransaction();
 
   /**
    * Pops the topmost transaction on the redo stack, calls its
    * redoTransaction() method, then pushes it on the undo stack.
    */
+  [can_run_script]
   void redoTransaction();
 
   /**
@@ -69,6 +72,7 @@
    * @param aData An arbitrary nsISupports object that is associated with the
    * batch. Can be retrieved from the undo or redo stacks.
    */
+  [can_run_script]
   void beginBatch(in nsISupports aData);
 
   /**